src/protocols/rdp/rdp.c

```diff
@@ -194,21 +247,11 @@ BOOL rdp_freerdp_pre_connect(freerdp* instance) {
     primary->MemBlt = guac_rdp_gdi_memblt;
     primary->OpaqueRect = guac_rdp_gdi_opaquerect;
 
-    /* Load "rdpgfx" plugin for Graphics Pipeline Extension */
-    if (settings->enable_gfx)
-        guac_rdp_rdpgfx_load_plugin(context);
-
-    /* Load plugin providing Dynamic Virtual Channel support, if required */
-    if (freerdp_settings_get_bool(GUAC_RDP_CONTEXT(instance)->settings, FreeRDP_SupportDynamicChannels) &&
-            guac_freerdp_channels_load_plugin(context, "drdynvc",
-                GUAC_RDP_CONTEXT(instance)->settings)) {
-        guac_client_log(client, GUAC_LOG_WARNING,
-                "Failed to load drdynvc plugin. Display update and audio "
-                "input support will be disabled.");
-    }
+    #ifndef RDP_INST_HAS_LOAD_CHANNELS
+        rdp_freerdp_load_channels(instance);
```

Comment:
Does this only run for FreeRDP2? The function docs say that, but it looks to me like it always runs?

Reply:
It does always run, but where it is called is changed depending on freerdp2 or 3. I'll update the docs to show that

Reply:
Err actually I think that this documentation is enough that I already had in the function comment?

>  * This function is designed to be used as the LoadChannels
>  * callback within FreeRDP3, but is also called directly from our pre_connect
>  * function in FreeRDP2 environments.

Reply:
This function is called directly from the pre_connect function _only_ in FreeRDP2 environments?

Reply:
Yeah

Reply:
Maybe I can reword to 

> 

        This function is designed to be used as the LoadChannels
        callback within FreeRDP3, or is called directly from our pre_connect
        function in FreeRDP2 environments.

Reply:
Oops, I misread this. Please ignore me.

Reply:
Eh, I think it's ok as is, but if you'd prefer to reword it, go for it.

Reply:
If it was misread then I probably wrote it not too clearly so I updated it

Reply:
The point of confusion was the relationship of the `RDP_INST_HAS_LOAD_CHANNELS` symbol to the FreeRDP version. It might be worth mentioning that it's only going to be defined for FreeRDP3, or just refer to what the configure check is actually checking, not the FreeRDP version itself, which isn't obvious at a first glance.

Reply:
Ahhh gotcha, added some comments next to each of the ifdefs

Reply:
That's better, but I think it might be worth just saying something about using this callback if available, not tying this behavior to a specific version that's not even checked in the configure check.

There's no guarantee that FreeRDP could release a 2.x version that _does_ have this callback if they wanted to, in which case this comment would be wrong.

Reply:
That looks better - mind updating the documentation for `rdp_freerdp_pre_connect()`, which still ties this to the FreeRDP version?

Reply:
Removed mentions of freerdp versions